### ISSUE TITLE
HELP NEEDED - Can you provide feedback on OCI SDKs?

### ISSUE THREAD

**Issue by kishore-oracle** [oracle/oci-typescript-sdk#24]
Are you an OCI customer and would you be willing to spend a few minutes on a call and provide feedback about your experience of using the OCI SDK? email: [kishore.ramamurthy@oracle.com](mailto:kishore.ramamurthy@oracle.com). Thanks.

**Comment by davidkhala** [comment]
I cannot find clear documentation on how to use oci-ts-sdk to interact with OCI Vault. Should it be `oci-vault` or `oci-keymanagement`? 

I also find out a lot of APIs have overhead notice
`NOTE: This class is auto generated by OracleSDKGenerator.`
So how and where should I contribute my code suggestion to these ts files?

**Comment by jodoglevy** [comment]
Hi @davidkhala for your first question - can you open a separate issue for that, and we can route that to the vault/kms teams?

For your second question, it is correct that any auto-generated file cannot be updated through a pull request. You can either file a github issue for the thing you'd like to be updated, or file a pull request with your suggested change, which we can use as a more precise example of what needs to be fixed/changed

**Comment by davidkhala** [comment]
> Hi @davidkhala for your first question - can you open a separate issue for that, and we can route that to the vault/kms teams?
> 
> For your second question, it is correct that any auto-generated file cannot be updated through a pull request. You can either file a github issue for the thing you'd like to be updated, or file a pull request with your suggested change, which we can use as a more precise example of what needs to be fixed/changed

I am afraid that the second is also a root cause of my Q1. Is the README.md of each sub-module also artifact of the code gen? 

And I wonder why it is designed to use code gen. I guess it can save time and effort for maintainer, but could be a nightmare for contributor.

**Comment by jodoglevy** [comment]
Yes, the README.md of each sub-module is also produced by the code gen.

Our goal is that OCI will be able to add/maintain everything customers need out of OCI SDKs, rather than requiring customers to own any required features / changes via contributions. Some files in the SDK are manually written, in which case a public contribution for those can be directly merged. For public contributions made to generated files, we consider those as a way for OCI to know exactly what changes a customer would like made to the SDK, and then we make the same change internally via the code generation process we use.

**Comment by jodoglevy** [comment]
@davidkhala can you open a github issue for the specific issue you need help with around WaaS? That way we can route it to the WaaS team to take a look

**Comment by smetal1** [comment]
I am trying to use this SDK for terraform resource manager stack creation , but there is no clear documentation or steps for it.

**Comment by jodoglevy** [comment]
Hi @smetal1 - i think it would be best for you to create a separate github issue for that, instead of us doing a back and forth on this github issue, which is not meant for discussing any particular issue.

Also, when you create the issue, please provide more info on the missing information which you'd like to get, so we may better assist

**Comment by roeycohen** [comment]
hi @kishore-oracle, some feedback for you :)

Although the TS types  are indeed helpful, the documentation is lacking, and I find myself many times creating an object in console and then fetching it in order to see how it is represented.

Also, it's not intuitive to understand which call requires what policy, and the online documentations are also sometimes lacking.
Leaving us with the option to "guess" and hope for the best.


Roey
